apis/v1alpha1/certificate_authority_activation.go

```diff
@@ -40,6 +40,30 @@ type CertificateAuthorityActivationSpec struct {
 	// This parameter must be supplied when you import a subordinate CA. When you
 	// import a root CA, there is no chain.
 	CertificateChain []byte `json:"certificateChain,omitempty"`
+	// Contains information to enable Online Certificate Status Protocol (OCSP)
+	// support, to enable a certificate revocation list (CRL), to enable both, or
+	// to enable neither. If this parameter is not supplied, existing capibilites
+	// remain unchanged. For more information, see the OcspConfiguration (https://docs.aws.amazon.com/privateca/latest/APIReference/API_OcspConfiguration.html)
+	// and CrlConfiguration (https://docs.aws.amazon.com/privateca/latest/APIReference/API_CrlConfiguration.html)
+	// types.
+	//
+	// The following requirements apply to revocation configurations.
+	//
+	//   - A configuration disabling CRLs or OCSP must contain only the Enabled=False
+	//     parameter, and will fail if other parameters such as CustomCname or ExpirationInDays
+	//     are included.
+	//
+	//   - In a CRL configuration, the S3BucketName parameter must conform to Amazon
+	//     S3 bucket naming rules (https://docs.aws.amazon.com/AmazonS3/latest/userguide/bucketnamingrules.html).
+	//
+	//   - A configuration containing a custom Canonical Name (CNAME) parameter
+	//     for CRLs or OCSP must conform to RFC2396 (https://www.ietf.org/rfc/rfc2396.txt)
+	//     restrictions on the use of special characters in a CNAME.
+	//
+	//   - In a CRL or OCSP configuration, the value of a CNAME parameter must
+	//     not include a protocol prefix such as "http://" or "https://".
+	RevocationConfiguration *RevocationConfiguration `json:"revocationConfiguration,omitempty"`
```

Comment:
Does it make sense to have the revocation configuration on the CAA and on the CA?

Reply:
Yes, we need it for CAA since its a field for the [UpdateCertificateAuthority api call](https://docs.aws.amazon.com/privateca/latest/APIReference/API_UpdateCertificateAuthority.html)

Reply:
Mmmm, but we just removed the update code path in https://github.com/aws-controllers-k8s/acmpca-controller/pull/11#discussion_r1446318885

Reply:
My point is, why do we wanna put a field in CA's spec if the user will not be able to update it at all?

Reply:
RevocationConfiguration is a field for the [CreateCertificateAuthority api cal](https://docs.aws.amazon.com/privateca/latest/APIReference/API_CreateCertificateAuthority.html)l as well, so we need it for the CA resource

Reply:
So users will create a revocation configure through CA creation but update it via the CAA?

Reply:
Yes